orttraining/orttraining/core/graph/training_op_defs.cc

```diff
@@ -1265,6 +1263,113 @@ void RegisterTrainingOpSchemas() {
           {"tensor(bool)"},
           "Constrain types to boolean tensors.");
 
+  /**
+   * AdamWOptimizer operator, taking multiple parameters as inputs (seq<tensor>).
+   * Ideally, a group of parameters sharing same learning rate (or other meta data) can use one single AdamWOptimizer.
+   * Implementation-wise, this bring opportunities for achieving better performance.
+   *
+   * The differences with AdamOptimizer:
+   * > Different inputs.
+   *
+   *   AdamWOptimizer can accept multiple parameters and other states related to them as inputs (seq<tensor>).
+   *   This make multi-tensor-apply applicable to the GPU implementation. Existing LambOptimizer has similar
+   *   capability, while it is using many fixed-length optional vardaric inputs, which is not a clean op definition.
+   *
+   *   AdamOptimizer takes one single parameter and its other states.
+   *
+   * > Different computation.
+   *
+   *   Despite of normal adam computation, for better perf in ORTTrainer, AdamOptimizer also fused gradient norm
+   *   clipping in its implementation. This sometimes makes it hard to align the optimizer with other frameworks during
+   *   model onboarding, on the other hand, the fusion did not bring very significant gains actually.
+   *
+   *   AdamWOptimizer has simplified definitions, excludes inputs/attributes not related to optimizer computations.
+   *
+   * AdamWOptimizer is recommended for new usage, AdamOptimizer is left as it is to support existing ORTTrainer
+   * solutions.
+   */
+  ONNX_CONTRIB_OPERATOR_SCHEMA(AdamWOptimizer)
+      .SetDomain(kMSDomain)
+      .SinceVersion(1)
+      .Input(0, "lr", "The initial learning rate.", "T1")
+      .Input(1, "step", "The update count of weights. It should be a scalar.", "T2")
```

Comment:
so step is not sent as output of the optimizer node? maybe that can be the required output instead of updated flag. also, is there a scenario where the updated flag will be False?

Reply:
My thinking is, step should be maintained by the users, just as they manage the changing learning rate. Returning the update step for user don't mean much for the users. 

If we don't need the optimizer apply gradient, we shouldn't let the execution goes to this OP (considering this OP takes no  gradient norm as input any more), we can manage this as well in Module/Optimizer train steps.

Reply:
Module/Optimizer don't have visibility over finiteness of gradient norm, because it is calculated in this graph itself. So just before this op there would be the grad norm calculation and we might need to have a no-op mode for this op. the updated flag can stay the same and optimizer can keep a count of steps

Reply:
In my opinion, the step should be maintained by the invoking optimizer class. The optimizer node only uses the step to calculate the bias correction, but it shouldn't know whether to increment the step by 1, or more or leave it as is. It should return whether the optimizer step was applied. Then based on this output, the optimizer class should maintain the step for the next execution.